src/urllib3/compat/rlock.py

```diff
@@ -0,0 +1,12 @@
+"""
+In this module, we shim out a context manager to be used as a compatibility
+layer if the system `threading` module doesn't have a real `RLock` available.
+"""
+
+
+class RLock:
```

Comment:
This could be replaced with `nullcontext` when dropping the project is Python3.7+. Perhaps a TODO comment could say as much.

https://docs.python.org/3/library/contextlib.html#contextlib.nullcontext

Reply:
You're definitely right, but if we're commenting all the TODOs we can make better once we're on newer versions of Python, there are going to be a lot of those. 😉

Reply:
Can this definition be placed elsewhere? I don't want to create new `urllib3.*` packages that aren't private to avoid confusing users on what our supported APIs are.

Reply:
@SethMichaelLarson, how would you feel about a `urllib3.compat._shims` module to put stuff like this?

Reply:
(btw my handle is @sethmlarson) and moving to `urllib3._compat` would be enough, the `_*` shows it's private same as `_collections`.

Reply:
My bad @sethmlarson! Autofill got me...